hera_cal/vis_clean.py

```diff
@@ -1672,95 +1672,47 @@ def _filter_argparser():
+    # clean arguments.
+    clean_options = ap.add_argument_group(title='Options for CLEAN (arguments only used if mode=="clean"!)')
     clean_options.add_argument("--window", type=str, default='blackman-harris', help='window function for frequency filtering (default "blackman-harris",\
                               see uvtools.dspec.gen_window for options')
     clean_options.add_argument("--maxiter", type=int, default=100, help='maximum iterations for aipy.deconv.clean to converge (default 100)')
     clean_options.add_argument("--edgecut_low", default=0, type=int, help="Number of channels to flag on lower band edge and exclude from window function.")
     clean_options.add_argument("--edgecut_hi", default=0, type=int, help="Number of channels to flag on upper band edge and exclude from window function.")
     clean_options.add_argument("--gain", type=float, default=0.1, help="Fraction of residual to use in each iteration.")
     clean_options.add_argument("--alpha", type=float, default=.5, help="If window='tukey', use this alpha parameter (default .5).")
-    return a
-
-# ------------------------------------------
-# Here are are parsers for linear filters.
-# for xtalk and foregrounds.
-# ------------------------------------------
-
-
-def _linear_argparser():
-    '''
-    Arg parser for commandline operation of hera_cal.delay_filter in various linear modes.
-    Arguments
-    ---------
-        None
-    Returns
-    -------
-        Arg-parser for linear filtering. Still needs domain specific args (delay versus xtalk)
-    '''
-    a = _filter_argparser()
-    cache_options = a.add_argument_group(title='Options for caching')
-    a.add_argument("--write_cache", default=False, action="store_true", help="if True, writes newly computed filter matrices to cache.")
-    a.add_argument("--cache_dir", type=str, default=None, help="directory to store cached filtering matrices in.")
-    a.add_argument("--read_cache", default=False, action="store_true", help="If true, read in cache files in directory specified by cache_dir.")
-    a.add_argument("--max_contiguous_edge_flags", type=int, default=1, help="Skip integrations with at least this number of contiguous edge flags.")
-    return a
-
-
-def _dpss_argparser():
-    '''
-    Arg parser for commandline operation of hera_cal.delay_filter in dpss mode.
-    Arguments
-    ---------
-        None
-    Returns
-    -------
-        Arg-parser for dpss filtering.
-    '''
-    a = _linear_argparser()
-    a.add_argument("--CLEAN_outfilename", default=None, type=str, help="path for writing the filtered model visibilities (with the same flags)")
-    a.add_argument("--filled_outfilename", default=None, type=str, help="path for writing the original data but with flags unflagged and replaced with filtered models wherever possible")
-    return a
+    cache_options = ap.add_argument_group(title='Options for caching (arguments only used if mode!="clean")')
```

Comment:
Can I suggest comments and whitespace between argument groups?

Reply:
I've added white space and comments.